app/controllers/fsapi_controller.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class FsapiController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate
+
+  def index
+    if params['section'] != 'dialplan'
+      logger.warn('Request from freeswitch mod_xml_curl for non-dialplan section')
+      return render 'reject'
+    end
+    if params['Caller-Destination-Number'].blank?
+      logger.warn('Missing Caller-Destination-Number')
+      return render 'reject'
+    end
+
+    @caller_dest_num = params['Caller-Destination-Number']
+    @pin = params['variable_pin']
+    @allotted_timeout = Rails.configuration.x.fsapi_max_duration * 60
+
+    # The variable 'pin' is set once the caller has gone through the prompt
+    # so if it's missing, send the meeting selection prompt.
+    if @pin.blank?
+      logger.info('Prompting for pin number')
+      return render 'pin_prompt'
+    end
+
+    # Look up the meeting for this voice bridge number
+    @meeting = begin
+      Meeting.find_by_voice_bridge(@pin)
+    rescue ApplicationRedisRecord::RecordNotFound
+      begin
+        # If no meeting was found, it might have been a breakout room pin, which adds one digit to the end
```

Comment:
BBB supports more than 10 breakout rooms, so this logic should check both `@pin[0...-1]` and `@pin[0...-2]` or be written as a loop.

Reply:
That's a good point. Perhaps rather than remove a fixed number of digits from the end, it would be better to try truncating the provided pin to the configured voice bridge length (if the pin is longer).

… But that wouldn't work if the "use external" option is enabled and the externally provided voice bridge numbers are shorter or longer than the configured length. Hmm.